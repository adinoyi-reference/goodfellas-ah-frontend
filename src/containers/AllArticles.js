import React, { Component } from 'react';
import { connect } from 'react-redux';
import Card from './ArticleCard';
import RecentStory from '../components/article/RecentStory';
import SecondArticleCard from '../components/article/SecondArticleCard';
import { getArticles } from '../actions/articleActions';
import dice from '../assets/dice.png';
import '../styles/styles.scss';


class AllArticles extends Component {
  componentWillMount = () => {
    const { getArticles: getAllArticles } = this.props;

    getAllArticles();
  }


  render() {
    return (
      <div>
        <div className="feed">
        Stories From Your Feed
          <img src={dice} alt="" />
        </div>
        <div className="hero-lowerbody-articles">
          <Card />
        </div>
        <hr className="card-hr" />
        <div className="second-card">
          <div className="all-articles-card">
            <div className="buttons">
              <button type="button">All Articles</button>
              <button type="button">Favourites</button>
            </div>

            <SecondArticleCard />
            <SecondArticleCard />

          </div>
          <div className="recent-stories">
            <div className="recent-stories-box">
              <p className="title">Recent Stories</p>

              <RecentStory />
              <RecentStory />

            </div>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  error: state.articles.error,
  articles: state.articles.articles,
});

export default connect(mapStateToProps, { getArticles })(AllArticles);
