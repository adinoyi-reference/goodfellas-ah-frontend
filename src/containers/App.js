import React from 'react';
import { BrowserRouter, Route } from 'react-router-dom';
import Hero from '../views/Hero';
import Signup from '../views/Signup';
import Signin from '../views/Signin';
import Profile from '../views/Profile';
import '../styles/styles.scss';

export default () => (
  <BrowserRouter>
    <div>
      <Route exact path="/" component={Hero} />
      <Route path="/auth/signin" render={(props) => <Signin formtype="signin" {...props} />} />
      <Route path="/auth/signup" render={(props) => <Signup formtype="signup" {...props} />} />
      <Route path="/user/profile" component={Profile} />
    </div>
  </BrowserRouter>
);
