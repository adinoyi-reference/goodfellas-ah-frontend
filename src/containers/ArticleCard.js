import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import parser from 'react-html-parser';
import { getArticles } from '../actions/articleActions';
import Loading from '../components/shared/Loading';
import { filterReactions } from '../mixin';

export class Card extends Component {
  state = {
    articleLimit: 6,
  };

  componentDidMount() {
    const { getArticles: getAllArticles } = this.props;
    getAllArticles();
  }

  sortArticlesByLikes = (articles) => {
    const sortedArticles = articles
      .map((eachArticle) => {
        const filteredArticles = filterReactions(eachArticle.reactions).likes;
        return { article: eachArticle, track: filteredArticles };
      })
      .sort((a, b) => {
        const result = b.track - a.track;
        return result;
      });
    return sortedArticles;
  };

  displayArticles = (articleLimit) => {
    const { articles } = this.props;
    const newlySortedArticles = this.sortArticlesByLikes(articles);
    const displayedArticles = newlySortedArticles.slice(0, 0 + articleLimit);
    return displayedArticles;
  };

  handleClick = (e) => {
    const { articleLimit } = this.state;
    if (articleLimit === 18) {
      e.currentTarget.innerHTML = '';
    }
    if (articleLimit !== 24) {
      this.setState({ articleLimit: articleLimit + 6 });
      this.displayCards(this.displayArticles);
    }
  };


  changeBackground = (image) => {
    const changeBackground = {
      backgroundImage: `linear-gradient(rgba(0, 0, 0, .32), rgba(0, 0, 0, .32)), url(${image}`,
      backgroundSize: 'cover',
    };
    return changeBackground;
  };

  displayCards = (displayArticles) => {
    const { articleLimit } = this.state;
    return displayArticles(articleLimit).map((card) => {
      const displayedDescription = card.article.description.slice(0, 120);
      const displayedTitle = card.article.title.slice(0, 24);
      return (
        <Link key={card.article.id} to={`/articles/${card.article.slug}`}>
          <div
            onClick={this.getArticle}
            className="hero-card"
          >
            <div className="hero-card-details col-sm-7">
              <h6>{parser(displayedTitle)}</h6>
              {parser(displayedDescription)}
              <div className="hero-card-author">
                {card.article.user.profile.image === null ? (
                  <img
                    src="https://res.cloudinary.com/drmmqcxkc/image/upload/v1541581893/Authors%20Haven/user-placeholder.png"
                    alt="Author Profile"
                  />
                ) : (
                  <img
                    src={card.article.user.profile.image}
                    alt="Author Profile"
                  />
                )}
                <p>
                  {card.article.user.firstname} {card.article.user.lastname}
                </p>
              </div>
            </div>
            {card.article.image === null ? (
              <div className="hero-card-image col-sm-5">
                <img
                  src="https://res.cloudinary.com/drmmqcxkc/image/upload/v1541426068/Authors%20Haven/icons8-heart-outline-48-grey.png"
                  alt=""
                />
                <p>{filterReactions(card.article.reactions).likes}</p>
              </div>
            ) : (
              <div
                style={this.changeBackground(card.article.image)}
                className="hero-card-image col-sm-5"
              >
                <img
                  src="https://res.cloudinary.com/drmmqcxkc/image/upload/v1541426068/Authors%20Haven/icons8-heart-outline-48-grey.png"
                  alt=""
                />
                <p>{filterReactions(card.article.reactions).likes}</p>
              </div>
            )}
          </div>
        </Link>
      );
    });
  };

  render() {
    const { articles } = this.props;
    return (
      <div>
        {articles ? (
          <div className="card-wrapper">
            {this.displayCards(this.displayArticles)}
            <div>
              {articles.length <= 6 ? (
                null
              ) : (
                <div onClick={this.handleClick} className="hero-moreArticles row">
                  <p ref="moreArticles">More Articles</p>
                  <img
                    src="https://res.cloudinary.com/drmmqcxkc/image/upload/v1541426068/Authors%20Haven/icons8-expand-arrow-24.png"
                    alt=""
                  />
                </div>
              )}
            </div>
          </div>
        ) : (
          <Loading />
        )}
      </div>
    );
  }
}

function mapStatetoProps(state) {
  return { articles: state.articles.articles.articles, error: state.articles.error.response };
}

export default connect(mapStatetoProps, { getArticles })(Card);
