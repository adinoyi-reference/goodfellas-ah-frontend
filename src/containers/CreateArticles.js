import React, { Component } from 'react';
import { connect } from 'react-redux';
import swal from 'sweetalert';
import Editor from 'react-medium-editor';
import 'medium-editor/dist/css/medium-editor.css';
import 'medium-editor/dist/css/themes/default.css';
import ImageUploader from '../components/articles/imageUpload';
import publishArticle from '../actions/publishArticle';
import '../styles/views/createArticles.scss';
import { Loader } from '../components/shared/Loading';


export class CreateArticles extends Component {
  state = {
    title: '',
    body: '',
    imageLoad: false,
    image: '',
  }

  componentDidUpdate(prevProps) {
    const { status, history, publishedArticle } = this.props;
    if (prevProps.status.success !== status.success && status.success) {
      swal('Good job!', 'Article Saved Successfully!', 'success');
      setTimeout(() => history.push(`/articles/${publishedArticle.article.slug}`), 3000);
    } else if (prevProps.status.error !== status.error && status.error) {
      swal('Error!', 'Something Went Wrong!', 'error');
    }
  }


  handleChange = (event) => {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  }

  generateImageTag = (url) => `<img class="inline-image" src="${url}" />`

  imageUploaded = (url) => {
    const imageHtml = this.generateImageTag(url);
    const { image } = this.state;
    this.setState((prevState) => ({ body: prevState.body + imageHtml, imageLoad: false }));

    if (image.length < 1) {
      this.setState({
        image: url,
      });
    }
  }

  handleSubmit = (event) => {
    const { history } = this.props;
    event.preventDefault();
    const { title, body } = this.state;
    if (!title || title === '<p><br></p>') return swal('Please add a title', 'Your article needs to have a title', 'warning');
    if (!body || body === '<p><br></p>') { return swal('Please add a body', 'The body of your article cannot be empty!', 'warning'); }
    const articlePayload = {
      ...this.state,
      description: body.split(' ').slice(0, 10).join(' '),
    };
    return history.push({
      pathname: '/tags',
      state: { articlePayload },
    });
  }

  render() {
    const handleEditorChange = (text, key) => this.setState({ [key]: text });
    const { imageUploadStatus } = this.props;
    const { title, body } = this.state;
    return (
      <div className="article-body">
        <div className="articles-card">
          <div className="article-buttons">
            <button className="btn article-whitebutton" type="submit" onClick={this.handleSubmit}>
            Publish
            </button>
          </div>
          <ImageUploader imageUploaded={this.imageUploaded} />

          <form>
            <Editor
              name="title"
              id="title"
              data-placeholder="Title"
              text={title}
              onChange={(e) => handleEditorChange(e, 'title')}
              options={{
                toolbar: {
                  buttons: ['bold', 'italic', 'underline', 'strikethrough', 'quote', 'anchor', 'h2', 'h3', 'orderedlist'],
                },
              }}
            />
            {imageUploadStatus.loading ? <Loader /> : (
              <Editor
                name="body"
                className="editable"
                data-placeholder="What would you like to talk about?"
                text={body}
                onChange={(e) => handleEditorChange(e, 'body')}
                options={{
                  toolbar: {
                    buttons: ['bold', 'italic', 'underline', 'strikethrough', 'quote', 'anchor', 'h2', 'h3', 'orderedlist'],
                  },
                }}
              />
            )}
          </form>

        </div>
        <div className="article-fixed-buttons">
          <button className="btn article-fixed-whitebutton" type="submit" onClick={this.handleSubmit}>
            Publish
          </button>
        </div>
      </div>
    );
  }
}


const mapStateToProps = ({
  imageUploadReducer: { status: imageUploadStatus },
  publishArticleReducer: { status, publishedArticle },
}) => ({
  status,
  publishedArticle,
  imageUploadStatus,
});

export default connect(mapStateToProps, { publishArticle })(CreateArticles);
