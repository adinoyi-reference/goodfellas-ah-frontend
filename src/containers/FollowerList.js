import React, { Fragment } from 'react';
import FollowCard from '../components/shared/FollowCard';
import Loading from '../components/shared/Loading';

const FollowerList = (props) => {
  if (props.loading) return <Loading />;
  return (
    <Fragment>
      {
        Array(5).fill().map((x, i) => <FollowCard key={i} type="follower" />)
      }
    </Fragment>);
};

export default FollowerList;
