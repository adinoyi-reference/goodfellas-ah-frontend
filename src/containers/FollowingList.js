import React, { Fragment } from 'react';
import FollowCard from '../components/shared/FollowCard';

const FollowingList = (props) => (
  <Fragment>
    {
      props.following.map((following) => <FollowCard key={following.id} type="following" user={following.followedUser} />)
    }
  </Fragment>);

export default FollowingList;
