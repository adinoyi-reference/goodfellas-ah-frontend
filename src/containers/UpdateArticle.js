import React, { Component } from 'react';
import { connect } from 'react-redux';
import swal from 'sweetalert';
import Editor from 'react-medium-editor';
import 'medium-editor/dist/css/medium-editor.css';
import 'medium-editor/dist/css/themes/default.css';
import ImageUploader from '../components/articles/imageUpload';
import updateArticle from '../actions/updateArticle';
import { getAnArticle } from '../actions/articleActions';
import '../styles/views/createArticles.scss';
import { Loader } from '../components/shared/Loading';
import icons from '../assets/icons.svg';


export class UpdateArticles extends Component {
  state = {
    title: '',
    body: '',
    imageLoad: false,
  }

  componentDidMount = () => {
    const { getAnArticle: fetchArticle, match } = this.props;
    fetchArticle(match.params.slug);
  }

  componentDidUpdate = (prevProps) => {
    const {
      status, article, history, match,
    } = this.props;
    if (prevProps.status.success !== status.success && status.success) {
      swal('Good job!', 'Article Updated Successfully!', 'success');
      setTimeout(() => history.push(`/articles/${match.params.slug}`), 3000);
    } else if (prevProps.status.error !== status.error && status.error) {
      swal('Error!', 'Something Went Wrong!', 'error');
    }
    if (prevProps.article !== article) this.setState({ title: article.title, body: article.body });
  }

  handleChange = (event) => {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  }

  generateImageTag = (url) => `<img class="inline-image" src="${url}" />`


  imageUploaded = (url) => {
    const imageHtml = this.generateImageTag(url);
    this.setState((prevState) => ({ body: prevState.body + imageHtml, imageLoad: false }));
  }

  handleSubmit = async (event) => {
    event.preventDefault();
    const { title, body } = this.state;
    if (!title || title === '<p><br></p>') return swal('Please add a title', 'Your article needs to have a title', 'warning');
    if (!body || body === '<p><br></p>') { return swal('Please add a body', 'The body of your article cannot be empty!', 'warning'); }
    const { match } = this.props;
    const articlePayload = {
      ...this.state,
      description: body.split(' ').slice(0, 10).join(' '),
    };
    // eslint-disable-next-line react/destructuring-assignment
    this.props.updateArticle(articlePayload, match.params.slug);
  }

  render() {
    const handleEditorChange = (text, key) => this.setState({ [key]: text });
    const { imageUploadStatus, articleLoading, articleError } = this.props;
    const { title, body } = this.state;
    if (articleLoading) return <Loader />;
    if (articleError) {
      return (
        <div className="no-record centralizer">
          <svg className="icon">
            <use xlinkHref={`${icons}#sad`} />
          </svg>&nbsp;&nbsp;
          <span>{articleError}</span>
        </div>
      );
    }
    return (

      <div className="container article-body">
        <div className="articles-card">
          <div className="article-buttons">
            <button className="btn article-whitebutton" type="submit" onClick={this.handleSubmit}>
            Update
            </button>
          </div>
          <ImageUploader imageUploaded={this.imageUploaded} />

          <form>
            <textarea id="title" value={title} onChange={(e) => handleEditorChange(e.target.value, 'title')} />
            {imageUploadStatus.loading ? <Loader /> : (
              <Editor
                name="body"
                className="editable"
                text={body}
                onChange={(e) => handleEditorChange(e, 'body')}
                options={{
                  placeholder: false,
                  toolbar: {
                    buttons: ['bold', 'italic', 'underline', 'strikethrough', 'quote', 'anchor', 'h2', 'h3', 'orderedlist'],
                  },
                }}
              />
            )}
          </form>

        </div>
        <div className="article-fixed-buttons">
          <button className="btn article-fixed-whitebutton" type="submit" onClick={this.handleSubmit}>
            Update
          </button>
        </div>
      </div>
    );
  }
}


const mapStateToProps = ({
  imageUploadReducer: { status: imageUploadStatus },
  updateArticleReducer: { status, updatedArticle },
  articles: { article, articleLoading, articleError },
}) => ({
  status,
  updatedArticle,
  imageUploadStatus,
  article,
  articleLoading,
  articleError,
});

export default connect(mapStateToProps, { updateArticle, getAnArticle })(UpdateArticles);
