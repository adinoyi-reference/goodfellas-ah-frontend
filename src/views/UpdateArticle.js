import React, { Component } from 'react';
import { connect } from 'react-redux';
import swal from 'sweetalert';
import Editor from 'react-medium-editor';
import 'medium-editor/dist/css/medium-editor.css';
import 'medium-editor/dist/css/themes/default.css';
import ImageUploader from '../components/articles/imageUpload';
import updateArticle from '../actions/updateArticle';
import '../styles/views/createArticles.scss';
import { Loader } from '../components/shared/Loading';
import Body from '../components/layout/DefaultLayout';

export class UpdateArticles extends Component {
  state = {
    title: '',
    body: '',
    imageLoad: false,
  }

  componentDidUpdate(prevProps) {
    const { status } = this.props;
    if (prevProps.status.success !== status.success && status.success) {
      swal('Good job!', 'Article Updated Successfully!', 'success');
    } else if (prevProps.status.error !== status.error && status.error) {
      swal('Error!', 'Something Went Wrong!', 'error');
    }
  }

  handleChange = (event) => {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  }

  generateImageTag = (url) => `<img class="inline-image" src="${url}" />`


  imageUploaded = (url) => {
    const imageHtml = this.generateImageTag(url);
    this.setState((prevState) => ({ body: prevState.body + imageHtml, imageLoad: false }));
  }

  handleSubmit = async (event) => {
    event.preventDefault();
    const { body } = this.state;
    const articlePayload = {
      ...this.state,
      description: body.split(' ').slice(0, 3).join(' '),
    };
    // eslint-disable-next-line react/destructuring-assignment
    this.props.updateArticle(articlePayload);
  }

  render() {
    const handleEditorChange = (text, key) => this.setState({ [key]: text });
    const { imageUploadStatus } = this.props;
    const { title, body } = this.state;
    return (
      <Body>
        <div className="article-body">
          <div className="articles-card">
            <div className="article-buttons">
              <button className="btn article-whitebutton" type="submit" onClick={this.handleSubmit}>
            Update
              </button>
            </div>
            <ImageUploader imageUploaded={this.imageUploaded} />

            <form>
              <Editor
                name="title"
                id="title"
                data-placeholder="Title"
                text={title}
                onChange={(e) => handleEditorChange(e, 'title')}
                options={{
                  toolbar: {
                    static: true,
                    sticky: false,
                    buttons: ['bold', 'italic', 'underline', 'strikethrough', 'quote', 'anchor', 'h2', 'h3', 'orderedlist'],
                    updateOnEmptySelection: true,
                  },
                }}
              />
              {imageUploadStatus.loading ? <Loader /> : (
                <Editor
                  name="body"
                  className="editable"
                  data-placeholder="What would you like to talk about?"
                  text={body}
                  onChange={(e) => handleEditorChange(e, 'body')}
                  options={{
                    toolbar: {
                      activeButtonClass: 'medium-editor-button-active',
                      allowMultiParagraphSelection: true,
                      static: true,
                      sticky: false,
                      buttons: ['bold', 'italic', 'underline', 'strikethrough', 'quote', 'anchor', 'h2', 'h3', 'orderedlist'],
                      updateOnEmptySelection: true,
                    },
                  }}
                />
              )}
            </form>

          </div>
          <div className="article-fixed-buttons">
            <button className="btn article-fixed-whitebutton" type="submit" onClick={this.handleSubmit}>
            Update
            </button>
          </div>
        </div>
      </Body>
    );
  }
}


const mapStateToProps = ({
  imageUploadReducer: { status: imageUploadStatus },
  updateArticleReducer: { status, updatedArticle },
}) => ({
  status,
  updatedArticle,
  imageUploadStatus,
});

export default connect(mapStateToProps, { updateArticle })(UpdateArticles);
