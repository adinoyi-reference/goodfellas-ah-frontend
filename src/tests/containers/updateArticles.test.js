import React from 'react';
import toJson from 'enzyme-to-json';
import { UpdateArticles } from '../../containers/UpdateArticle';

describe('update artcicle container test suite', () => {
  const props = {
    status: {
      error: false,
      success: false,
    },
    updatedArticle: {},
    imageUploadStatus: {
      error: false,
      success: false,
    },
    updateArticle: jest.fn(),
    getAnArticle: jest.fn(),
  };
  const wrapper = shallow(<UpdateArticles {...props} match={{ params: { slug: 'we-are-here' } }} />);


  it('renders properly when mounted', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should not dispatch updateArticle when the update button is clicked without title or body', () => {
    const updateButton = wrapper.find('.article-whitebutton').at(0);
    updateButton.simulate('click', { preventDefault: jest.fn() });
    expect(props.updateArticle).not.toBeCalled();
    props.updateArticle.mockRestore();
  });
  it('dispatches updateArticle when the update button is clicked', () => {
    const updateButton = wrapper.find('.article-whitebutton').at(0);
    wrapper.setState({ title: 'some title', body: 'somebody' });
    updateButton.simulate('click', { preventDefault: jest.fn() });
    expect(props.updateArticle).toBeCalled();
    props.updateArticle.mockRestore();
  });
  describe('render', () => {
    test('container should render as expected', () => {
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
    });
  });
});
