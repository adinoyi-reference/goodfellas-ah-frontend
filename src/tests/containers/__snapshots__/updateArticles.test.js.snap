// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`update artcicle container test suite render container should render as expected 1`] = `
<div
  className="container article-body"
>
  <div
    className="articles-card"
  >
    <div
      className="article-buttons"
    >
      <button
        className="btn article-whitebutton"
        onClick={[Function]}
        type="submit"
      >
        Update
      </button>
    </div>
    <Connect(ImageUploader)
      imageUploaded={[Function]}
    />
    <form>
      <textarea
        id="title"
        onChange={[Function]}
        value="some title"
      />
      <ReactMediumEditor
        className="editable"
        name="body"
        onChange={[Function]}
        options={
          Object {
            "placeholder": false,
            "toolbar": Object {
              "buttons": Array [
                "bold",
                "italic",
                "underline",
                "strikethrough",
                "quote",
                "anchor",
                "h2",
                "h3",
                "orderedlist",
              ],
            },
          }
        }
        tag="div"
        text="somebody"
      />
    </form>
  </div>
  <div
    className="article-fixed-buttons"
  >
    <button
      className="btn article-fixed-whitebutton"
      onClick={[Function]}
      type="submit"
    >
      Update
    </button>
  </div>
</div>
`;

exports[`update artcicle container test suite renders properly when mounted 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UpdateArticles
    getAnArticle={
      [MockFunction] {
        "calls": Array [
          Array [
            "we-are-here",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    imageUploadStatus={
      Object {
        "error": false,
        "success": false,
      }
    }
    match={
      Object {
        "params": Object {
          "slug": "we-are-here",
        },
      }
    }
    status={
      Object {
        "error": false,
        "success": false,
      }
    }
    updateArticle={[MockFunction]}
    updatedArticle={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="articles-card"
        >
          <div
            className="article-buttons"
          >
            <button
              className="btn article-whitebutton"
              onClick={[Function]}
              type="submit"
            >
              Update
            </button>
          </div>
          <Connect(ImageUploader)
            imageUploaded={[Function]}
          />
          <form>
            <textarea
              id="title"
              onChange={[Function]}
              value=""
            />
            <ReactMediumEditor
              className="editable"
              name="body"
              onChange={[Function]}
              options={
                Object {
                  "placeholder": false,
                  "toolbar": Object {
                    "buttons": Array [
                      "bold",
                      "italic",
                      "underline",
                      "strikethrough",
                      "quote",
                      "anchor",
                      "h2",
                      "h3",
                      "orderedlist",
                    ],
                  },
                }
              }
              tag="div"
              text=""
            />
          </form>
        </div>,
        <div
          className="article-fixed-buttons"
        >
          <button
            className="btn article-fixed-whitebutton"
            onClick={[Function]}
            type="submit"
          >
            Update
          </button>
        </div>,
      ],
      "className": "container article-body",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="article-buttons"
            >
              <button
                className="btn article-whitebutton"
                onClick={[Function]}
                type="submit"
              >
                Update
              </button>
            </div>,
            <Connect(ImageUploader)
              imageUploaded={[Function]}
            />,
            <form>
              <textarea
                id="title"
                onChange={[Function]}
                value=""
              />
              <ReactMediumEditor
                className="editable"
                name="body"
                onChange={[Function]}
                options={
                  Object {
                    "placeholder": false,
                    "toolbar": Object {
                      "buttons": Array [
                        "bold",
                        "italic",
                        "underline",
                        "strikethrough",
                        "quote",
                        "anchor",
                        "h2",
                        "h3",
                        "orderedlist",
                      ],
                    },
                  }
                }
                tag="div"
                text=""
              />
            </form>,
          ],
          "className": "articles-card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="btn article-whitebutton"
                onClick={[Function]}
                type="submit"
              >
                Update
              </button>,
              "className": "article-buttons",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Update",
                "className": "btn article-whitebutton",
                "onClick": [Function],
                "type": "submit",
              },
              "ref": null,
              "rendered": "Update",
              "type": "button",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "imageUploaded": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <textarea
                  id="title"
                  onChange={[Function]}
                  value=""
                />,
                <ReactMediumEditor
                  className="editable"
                  name="body"
                  onChange={[Function]}
                  options={
                    Object {
                      "placeholder": false,
                      "toolbar": Object {
                        "buttons": Array [
                          "bold",
                          "italic",
                          "underline",
                          "strikethrough",
                          "quote",
                          "anchor",
                          "h2",
                          "h3",
                          "orderedlist",
                        ],
                      },
                    }
                  }
                  tag="div"
                  text=""
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "id": "title",
                  "onChange": [Function],
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "textarea",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "editable",
                  "name": "body",
                  "onChange": [Function],
                  "options": Object {
                    "placeholder": false,
                    "toolbar": Object {
                      "buttons": Array [
                        "bold",
                        "italic",
                        "underline",
                        "strikethrough",
                        "quote",
                        "anchor",
                        "h2",
                        "h3",
                        "orderedlist",
                      ],
                    },
                  },
                  "tag": "div",
                  "text": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "form",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="btn article-fixed-whitebutton"
            onClick={[Function]}
            type="submit"
          >
            Update
          </button>,
          "className": "article-fixed-buttons",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Update",
            "className": "btn article-fixed-whitebutton",
            "onClick": [Function],
            "type": "submit",
          },
          "ref": null,
          "rendered": "Update",
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="articles-card"
          >
            <div
              className="article-buttons"
            >
              <button
                className="btn article-whitebutton"
                onClick={[Function]}
                type="submit"
              >
                Update
              </button>
            </div>
            <Connect(ImageUploader)
              imageUploaded={[Function]}
            />
            <form>
              <textarea
                id="title"
                onChange={[Function]}
                value=""
              />
              <ReactMediumEditor
                className="editable"
                name="body"
                onChange={[Function]}
                options={
                  Object {
                    "placeholder": false,
                    "toolbar": Object {
                      "buttons": Array [
                        "bold",
                        "italic",
                        "underline",
                        "strikethrough",
                        "quote",
                        "anchor",
                        "h2",
                        "h3",
                        "orderedlist",
                      ],
                    },
                  }
                }
                tag="div"
                text=""
              />
            </form>
          </div>,
          <div
            className="article-fixed-buttons"
          >
            <button
              className="btn article-fixed-whitebutton"
              onClick={[Function]}
              type="submit"
            >
              Update
            </button>
          </div>,
        ],
        "className": "container article-body",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="article-buttons"
              >
                <button
                  className="btn article-whitebutton"
                  onClick={[Function]}
                  type="submit"
                >
                  Update
                </button>
              </div>,
              <Connect(ImageUploader)
                imageUploaded={[Function]}
              />,
              <form>
                <textarea
                  id="title"
                  onChange={[Function]}
                  value=""
                />
                <ReactMediumEditor
                  className="editable"
                  name="body"
                  onChange={[Function]}
                  options={
                    Object {
                      "placeholder": false,
                      "toolbar": Object {
                        "buttons": Array [
                          "bold",
                          "italic",
                          "underline",
                          "strikethrough",
                          "quote",
                          "anchor",
                          "h2",
                          "h3",
                          "orderedlist",
                        ],
                      },
                    }
                  }
                  tag="div"
                  text=""
                />
              </form>,
            ],
            "className": "articles-card",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="btn article-whitebutton"
                  onClick={[Function]}
                  type="submit"
                >
                  Update
                </button>,
                "className": "article-buttons",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Update",
                  "className": "btn article-whitebutton",
                  "onClick": [Function],
                  "type": "submit",
                },
                "ref": null,
                "rendered": "Update",
                "type": "button",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "imageUploaded": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <textarea
                    id="title"
                    onChange={[Function]}
                    value=""
                  />,
                  <ReactMediumEditor
                    className="editable"
                    name="body"
                    onChange={[Function]}
                    options={
                      Object {
                        "placeholder": false,
                        "toolbar": Object {
                          "buttons": Array [
                            "bold",
                            "italic",
                            "underline",
                            "strikethrough",
                            "quote",
                            "anchor",
                            "h2",
                            "h3",
                            "orderedlist",
                          ],
                        },
                      }
                    }
                    tag="div"
                    text=""
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "id": "title",
                    "onChange": [Function],
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "textarea",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "className": "editable",
                    "name": "body",
                    "onChange": [Function],
                    "options": Object {
                      "placeholder": false,
                      "toolbar": Object {
                        "buttons": Array [
                          "bold",
                          "italic",
                          "underline",
                          "strikethrough",
                          "quote",
                          "anchor",
                          "h2",
                          "h3",
                          "orderedlist",
                        ],
                      },
                    },
                    "tag": "div",
                    "text": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "form",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="btn article-fixed-whitebutton"
              onClick={[Function]}
              type="submit"
            >
              Update
            </button>,
            "className": "article-fixed-buttons",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Update",
              "className": "btn article-fixed-whitebutton",
              "onClick": [Function],
              "type": "submit",
            },
            "ref": null,
            "rendered": "Update",
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
